spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -1455,6 +1490,37 @@ private void fixTxOffsetsIfNeeded() {
 			}
 		}
 
+		private void beforePoll() {
+			if (this.isBatchListener) {
+				batchInterceptBeforePoll();
+			}
+			else {
+				recordInterceptBeforePoll();
+			}
```

Comment:
Sounds more like there has to be a common Before-After contract, but not the same copy in record and batch interceptors.